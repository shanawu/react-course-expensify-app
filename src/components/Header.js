import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => {
  return (
    <header>
      <h1>Expensify</h1>
      <NavLink
        to="/"
        className={({ isActive }) => (isActive ? "active-link" : "")}
      >
        Home
      </NavLink>
      <NavLink
        to="/create"
        className={({ isActive }) => (isActive ? "active-link" : "")}
      >
        Add Expense
      </NavLink>
      <NavLink
        to="/edit"
        className={({ isActive }) => (isActive ? "active-link" : "")}
      >
        Edit Expense
      </NavLink>
      <NavLink
        to="/help"
        className={({ isActive }) => (isActive ? "active-link" : "")}
      >
        Help
      </NavLink>
    </header>
  );
};

export default Header;
