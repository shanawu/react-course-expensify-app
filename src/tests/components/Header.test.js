import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import Header from "../../components/Header";

test("should render Header", () => {
  const { container } = render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );

  expect(container).toMatchSnapshot();
});

// import React from "react";
// import { render, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
// import { BrowserRouter } from "react-router-dom";
// import Header from "../../components/Header";

// test("should render Header with correct navigation links", () => {
//   render(
//     <BrowserRouter>
//       <Header />
//     </BrowserRouter>
//   );

//   // If you're using `screen`, then skip the container argument:

//   const homeLink = screen.getByText("Home"); // search for all elements that have the text - "Home"
//   expect(homeLink).toBeInTheDocument(); // if the homeLink is in the body of the document
//   expect(homeLink.href).toContain("/"); // if the homeLink is correct - "/"

//   const addExpenseLink = screen.getByText("Add Expense");
//   expect(addExpenseLink).toBeInTheDocument();
//   expect(addExpenseLink.href).toContain("/create"); // if the addExpenseLink is correct - "/create"

//   const editExpenseLink = screen.getByText("Edit Expense");
//   expect(editExpenseLink).toBeInTheDocument();
//   expect(editExpenseLink.href).toContain("/edit"); // if the addExpenseLink is correct - "/edit"

//   const helpLink = screen.getByText("Help");
//   expect(helpLink).toBeInTheDocument();
//   expect(helpLink.href).toContain("/help"); // if the addExpenseLink is correct - "/help"
// });
