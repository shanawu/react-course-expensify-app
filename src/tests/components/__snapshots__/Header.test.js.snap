// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<div>
  <header>
    <h1>
      Expensify
    </h1>
    <a
      aria-current="page"
      class="active-link"
      href="/"
    >
      Home
    </a>
    <a
      class=""
      href="/create"
    >
      Add Expense
    </a>
    <a
      class=""
      href="/edit"
    >
      Edit Expense
    </a>
    <a
      class=""
      href="/help"
    >
      Help
    </a>
  </header>
</div>
`;
